src/box/memtx_tree.cc

```diff
@@ -611,15 +634,38 @@ tree_iterator_start(struct iterator *iterator, struct tuple **ret)
 		 * last position in the tree, that's what we need.
 		 */
 		memtx_tree_iterator_prev(tree, &it->tree_iterator);
+		res = memtx_tree_iterator_get_elem(tree, &it->tree_iterator);
+	}
+
+	/*
+	 * We need to use value semantics here, as the result's underlying data
+	 * can change after we cross the “MVCC TRANSACTION MANAGER STORY GARBAGE
+	 * COLLECTION BOUND”.
+	 */
+	struct memtx_tree_data<USE_HINT> res_val;
+	if (res != NULL) {
```

Comment:
Nit: afaik one-like body is not enclosed with braces

Reply:
I would argue about this: in my experience, there were several times, when I encountered a bug, caused by missing if-block braces, and the assumption, that all the indented lines after the if will get executed — I would rather be explicit and verbose. 

> Better safe than sorry.

Reply:
We should not argue about code style in PR. There's a [directive](https://www.tarantool.io/en/doc/latest/dev_guide/c_style_guide/):
> Do not unnecessarily use braces where a single statement will do.

Please respect and fix.

Reply:
Fixed.